### DISCUSSION TITLE
Notification doesn't work in self hosted server

### DISCUSSION THREAD

**Discussion by perperj-super** [hcengineering/platform#5651] [Q&A] [answered]
Hello!

My name is James and I'm interested in this project.
I set up the local huly server using [huly-selfhost](https://github.com/hcengineering/huly-selfhost) and it was quite successful.
But I noticed that notification doesn't work for me, tried enable all notification settings, but nothing worked for me.

Could anybody help me on this?

Thanks!

**Comment by BykhovDenis** [comment, marked as answer]
Hello
For push notification you should follow these guidelines:

1. Ensure the service operates exclusively over HTTPS.

Generate VAPID keys and provide them as environment variables to your containers:

For the front container, set:
PUSH_PUBLIC_KEY
For the transactor container, set:
PUSH_PUBLIC_KEY
PUSH_PRIVATE_KEY
PUSH_SUBJECT (formatted as mailto:${mailUsedForKeysGeneration})

Email notification currently not available for self-hosting solution
